### PR TITLE
Prepare `Locator` for async migration

### PR DESCRIPTION
**Pull request by inancgumus**
## What?

Turns panics into errors.

## Why?

To make the async migration more reliable.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests for my changes
- [x] I have commented on my code, particularly in hard-to-understand areas

## Related PR(s)/Issue(s)

#1305

### CODE REVIEW

File: tests/locator_test.go [hunk 1]
```diff
@@ -469,7 +419,7 @@ func TestLocator(t *testing.T) { //nolint:gocognit
 			tb := newTestBrowser(t)
 			p := tb.NewPage(nil)
 			p.SetContent("<html></html>", nil)
-			assert.Panics(t, func() { tt.do(p.Locator("NOTEXIST", nil), tb) })
+			require.Error(t, tt.do(p.Locator("NOTEXIST", nil), tb))
```
Comment (ankur22) [inline tests/locator_test.go:422]: Woohoo! Nice 😄

**Review by ankur22 — APPROVED** [review]
LGTM

File: tests/locator_test.go (hunk 1 above)
Comment (inancgumus) [reply tests/locator_test.go:422]: Yeah, we finally got rid of it 🥳

File: tests/locator_test.go [hunk 2]
```diff
@@ -442,7 +442,7 @@ func TestLocatorElementState(t *testing.T) {
 		{
 			"readOnly",
 			`() => document.getElementById('inputText').readOnly = true`,
-			func(l *common.Locator) bool { return l.IsEditable(nil) },
+			func(l *common.Locator) bool { resp, _ := l.IsEditable(nil); return resp },
```
[the remark this answers is not in the document]
Comment (inancgumus) [reply tests/locator_test.go:472]: Let's do that! 8e45ae64

**Review by ankur22 — APPROVED** [review]

**Pull request merged**